src/state/mod.rs

```diff
@@ -31,28 +32,108 @@ fn default_true() -> bool {
     true
 }
 
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct ModGroup {
+    pub mods: Vec<ModConfig>,
+}
+
 #[obake::versioned]
 #[obake(version("0.0.0"))]
+#[obake(version("0.1.0"))]
 #[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
 pub struct ModProfile {
     #[obake(cfg("0.0.0"))]
     pub mods: Vec<ModConfig>,
+
+    /// Mapping between mod groups and if they are enabled for a profile.
+    #[obake(cfg("0.1.0"))]
+    pub mod_groups: IndexMap<String, bool>,
+}
+
+impl From<ModProfile!["0.0.0"]> for ModProfile!["0.1.0"] {
+    fn from(_legacy: ModProfile!["0.0.0"]) -> Self {
+        // The migration requires `ModData` to handle instead.
+        unimplemented!("migration requires handling from `ModData`")
+    }
 }
 
 #[obake::versioned]
 #[obake(version("0.0.0"))]
+#[obake(version("0.1.0"))]
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub struct ModData {
     pub active_profile: String,
     #[obake(cfg("0.0.0"))]
     pub profiles: BTreeMap<String, ModProfile!["0.0.0"]>,
+    #[obake(cfg("0.1.0"))]
+    pub profiles: BTreeMap<String, ModProfile!["0.1.0"]>,
+    #[obake(cfg("0.1.0"))]
+    pub active_group: String,
```

Comment:
I think `active_group` shouldn't be a member persisted to state but instead derived from `active_profile`. If `active_profile` were *just* for UI I would say it shouldn't be persisted either, but it's used to signify which profile is installed/should be installed. Groups are only members of profiles and aren't directly installed so they shouldn't individually have a sense of "activeness".

Reply:
Removed active_group.